plugins/inputs/tengine/tengine.go

```diff
@@ -0,0 +1,335 @@
+}
+
+func (n *Tengine) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	// Create an HTTP client that is re-used for each
+	// collection interval
+	if n.client == nil {
+		client, err := n.createHttpClient()
+		if err != nil {
+			return err
+		}
+		n.client = client
+	}
+
+	for _, u := range n.Urls {
+		addr, err := url.Parse(u)
+		if err != nil {
+			acc.AddError(fmt.Errorf("Unable to parse address '%s': %s", u, err))
+			continue
+		}
+
+		wg.Add(1)
+		go func(addr *url.URL) {
+			defer wg.Done()
+			acc.AddError(n.gatherUrl(addr, acc))
+		}(addr)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (n *Tengine) createHttpClient() (*http.Client, error) {
+	tlsCfg, err := n.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	if n.ResponseTimeout.Duration < time.Second {
+		n.ResponseTimeout.Duration = time.Second * 5
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+		},
+		Timeout: n.ResponseTimeout.Duration,
+	}
+
+	return client, nil
+}
+type TengineSatus struct {
+	host string `json:"host"`
+	bytes_in uint64 `json:"bytes_in"`
+	bytes_out uint64 `json:"bytes_out"`
+	conn_total uint64 `json:"conn_total"`
+	req_total uint64 `json:"req_total"`
+	http_2xx uint64 `json:"http_2xx"`
+	http_3xx uint64 `json:"http_3xx"`
+	http_4xx uint64 `json:"http_4xx"`
+	http_5xx uint64 `json:"http_5xx"`
+	http_other_status uint64 `json:"http_other_status"`
+	rt uint64 `json:"rt"`
+	ups_req uint64 `json:"ups_req"`
+	ups_rt uint64 `json:"ups_rt"`
+	ups_tries uint64 `json:"ups_tries"`
+	http_200 uint64 `json:"http_200"`
+	http_206 uint64 `json:"http_206"`
+	http_302 uint64 `json:"http_302"`
+	http_304 uint64 `json:"http_304"`
+	http_403 uint64 `json:"http_403"`
+	http_404 uint64 `json:"http_404"`
+	http_416 uint64 `json:"http_416"`
+	http_499 uint64 `json:"http_499"`
+	http_500 uint64 `json:"http_500"`
+	http_502 uint64 `json:"http_502"`
+	http_503 uint64 `json:"http_503"`
+	http_504 uint64 `json:"http_504"`
+	http_508 uint64 `json:"http_508"`
+	http_other_detail_status uint64 `json:"http_other_detail_status"`
+	http_ups_4xx uint64 `json:"http_ups_4xx"`
+	http_ups_5xx uint64 `json:"http_ups_5xx"`
+}
+
+func (n *Tengine) gatherUrl(addr *url.URL, acc telegraf.Accumulator) error {
+	var tenginestatus TengineSatus
+	resp, err := n.client.Get(addr.String())
+	if err != nil {
+		return fmt.Errorf("error making HTTP request to %s: %s", addr.String(), err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("%s returned HTTP status %s", addr.String(), resp.Status)
+	}
+	r := bufio.NewReader(resp.Body)
+
+	for {
+		line, err := r.ReadString('\n')
+
+		if err != nil || io.EOF == err {
+			break
+		}
+		line_split := strings.Split(strings.TrimSpace(line), ",")
+		tenginestatus.host= line_split[0]
+		if err != nil {
+			return err
+		}
+		tenginestatus.bytes_in, err = strconv.ParseUint(line_split[1], 10, 64)
```

Comment:
We probably need to fix this in the nginx plugin, but ensure that the line is split as expected by checking the length first.

Reply:
OK, I have fixed it